libcnb/src/layer/public_interface.rs

```diff
@@ -10,6 +10,94 @@ use serde::Serialize;
 use std::collections::HashMap;
 use std::path::{Path, PathBuf};
 
+/// A specialized version of [`LayerTypes`](LayerTypes)
+/// where [`cached`](LayerTypes.cached) is always true.
+#[derive(Debug, Default, Eq, PartialEq)]
+pub struct CachedLayerTypes {
+    pub build: bool,
+    pub launch: bool,
+}
+/// Represents a buildpack layer that uses `cache: true` with the libcnb framework.
+///
+/// Use of this layer trait enforces that `existing_layer_strategy` is implemented.
+///
+/// Buildpack authors implement this trait to define how a layer is created/updated/removed
+/// depending on its state. To use a `Layer` implementation during build, use
+/// [`BuildContext::handle_layer`](crate::build::BuildContext::handle_layer).
+#[allow(unused_variables)]
+pub trait CachedLayer {
+    type Buildpack: Buildpack;
+
+    type Metadata: DeserializeOwned + Serialize + Clone;
+
+    /// See [`types`](Layer::types)
+    fn types(&self) -> CachedLayerTypes;
+
+    /// See [`create`](Layer::create)
+    fn create(
+        &self,
+        context: &BuildContext<Self::Buildpack>,
+        layer_path: &Path,
+    ) -> Result<LayerResult<Self::Metadata>, <Self::Buildpack as Buildpack>::Error>;
+
+    /// See [`create`](Layer::existing_layer_strategy)
+    fn existing_layer_strategy(
+        &self,
+        context: &BuildContext<Self::Buildpack>,
+        layer_data: &LayerData<Self::Metadata>,
+    ) -> Result<ExistingLayerStrategy, <Self::Buildpack as Buildpack>::Error>;
+
+    fn update(
+        &self,
+        context: &BuildContext<Self::Buildpack>,
+        layer_data: &LayerData<Self::Metadata>,
+    ) -> Result<LayerResult<Self::Metadata>, <Self::Buildpack as Buildpack>::Error> {
+        // I couldn't figure out how to allow defining this method without making it required
+        // and still defer to the original default implementation in `Layer`.
```

Comment:
I couldn't figure out how to default this to the default Layer implementation of `update`. I'm not sure it's possible. My one thought was to make some function like `default_update` and refactor `LayerType::update` to use it internally, and then use it here as well.

I'm open to other suggestions, or maybe unimplemented here is actually okay?

Same issue with `migrate_incompatible_metadata`